modules/Core/pages/panel/users_integrations.php

```diff
@@ -0,0 +1,259 @@
+            'icon' => Output::getClean($integration->getIcon()),
+            'link' => URL::build('/panel/users/integrations/', 'id=' . $view_user->data()->id . '&action=link&integration=' . $integration->getName()),
+            'edit' => URL::build('/panel/users/integrations/', 'id=' . $view_user->data()->id . '&action=edit&integration=' . $integration->getName())
+        ];
+    }
+
+    $user_integrations_list = [];
+    foreach ($view_user->getIntegrations() as $key => $integrationUser) {
+        if ($integrationUser->data()->username == null && $integrationUser->data()->identifier == null) {
+            continue;
+        }
+
+        $user_integrations_list[$key] = [
+            'identifier' => Output::getClean($integrationUser->data()->identifier),
+            'username' => Output::getClean($integrationUser->data()->username),
+            'verified' => Output::getClean($integrationUser->isVerified())
+        ];
+    }
+
+    $smarty->assign([
+        'VIEWING_USER_INTEGRATIONS' => str_replace('{user}', Output::getClean($view_user->data()->username), $language->get('admin', 'viewing_integrations_for_x')),
+        'INTEGRATION' => $language->get('admin', 'integration'),
+        'INTEGRATIONS' => $integrations_list,
+        'USER_INTEGRATIONS' => $user_integrations_list,
+        'MANUAL_LINKING' => $language->get('admin', 'manual_linking'),
+        'EDIT' => $language->get('general', 'edit'),
+        'NOT_LINKED' => $language->get('admin', 'not_linked'),
+        'ARE_YOU_SURE' => $language->get('general', 'are_you_sure'),
+        'ARE_YOU_SURE_MESSAGE' => $language->get('admin', 'unlink_account_confirm'),
+        'YES' => $language->get('general', 'yes'),
+        'NO' => $language->get('general', 'no'),
+        'UNLINK' => $language->get('admin', 'unlink'),
+        'BACK_LINK' => URL::build('/panel/user/' . Output::getClean($view_user->data()->id . '-' . $view_user->data()->username)),
+        'UNLINK_LINK' => URL::build('/panel/users/integrations/', 'id=' . $view_user->data()->id . '&action=unlink&integration='),
+    ]);
+
+    $template_file = 'core/users_integrations.tpl';
+} else if (isset($_GET['integration'])) {
+    switch ($_GET['action']) {
+        case 'link':
+            // Manual linking to integration (Integration User might already exist due of pending completion)
+            $integration = Integrations::getInstance()->getIntegration($_GET['integration']);
+            if ($integration == null) {
+                Redirect::to(URL::build('/panel/users/integrations/', 'id=' . $view_user->data()->id));
+            }
+
+            $integrationUser = $view_user->getIntegration($_GET['integration']);
+            if ($integrationUser != null && $integrationUser->data()->username != null && $integrationUser->data()->identifier != null) {
+                Redirect::to(URL::build('/panel/users/integrations/', 'id=' . $view_user->data()->id));
+            }
+
+            if (Input::exists()) {
```

Comment:
each of the actions in this switch statement check `Input::exists()` and `Token::check()`, it would be cleaner to check both of those before the switch statement starts